### PR TITLE
chore: use workflow from the `scripts` repository

### PR DESCRIPTION
**Pull request by Panquesito7**
Things added/changed:

- Use workflow from the `scripts` repository.

### CODE REVIEW

**Review by vzvu3k6k — COMMENTED** [review]
@Panquesito7 Thank you!

I've confirmed that the action is working fine already in [TheAlgorithms/C-Plus-Plus](https://github.com/TheAlgorithms/C-Plus-Plus/blob/2d492834b17fe578e5e624bc7a94ae21ad1d2d49/.github/workflows/directory_writer.yml), and also confirmed that [`python build_directory_md.py Ruby . .rb`](https://github.com/TheAlgorithms/scripts/blob/main/directory_md/build_directory_md.py)  generates a valid `DIRECTORY.md`.

I think it's almost fine, but I think there is a need to fix on.push.branches. Can you please check the comment?

File: .github/workflows/update_directory_md.yml [hunk 1]
```diff
@@ -1,59 +1,30 @@
-name: update_directory_md
-on: [push]
+name: Directory writer
+on:
+  push:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (Panquesito7) [reply .github/workflows/update_directory_md.yml:5]: Ahh yes, you're right, my mistake. Let's keep it `master` for now to prevent issues. 🙂

**Review by vzvu3k6k — APPROVED** [review]
Thank you!

**Comment by vzvu3k6k** [comment]
@Panquesito7 It seems I missed something during the review...

The action failed after the merge ([log](https://github.com/TheAlgorithms/Ruby/actions/runs/5337370931/jobs/9673337632)). So this PR has been temporarily reverted by #211.

I would appreciate it if you could submit a pull request with the bug fixed. I will investigate when I have time as well.

(By the way, I checked the TheAlgorithms/C-Plus-Plus repository again, and it appears that this action is not actually executed, because [the branch name of the action `on.push.branches`](https://github.com/TheAlgorithms/C-Plus-Plus/blob/2d492834b17fe578e5e624bc7a94ae21ad1d2d49/.github/workflows/directory_writer.yml) and the default branch are different.)

**Comment by Panquesito7** [comment]
Thanks for spotting that out! Weird that happened 👀 I believe the issue should be made in the scripts repository.
Would you like to report it or should I do it myself? Thanks. 🙂

**Comment by vzvu3k6k** [comment]
@Panquesito7 Thanks for the confirmation! I have reported it in https://github.com/TheAlgorithms/scripts/issues/27.

For the TheAlgorithms/C-Plus-Plus, I'm not sure whether to report the issue in an existing issue or create a new one. I'll leave the decision up to you.

Anyway I think this workflow would be very useful if it works. Thank you for trying to develop it :+1:

File: .github/workflows/update_directory_md.yml [hunk 2]
```diff
@@ -1,59 +1,30 @@
...
-          git config --global user.email '${GITHUB_ACTOR}@users.noreply.github.com'
-          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
-          git add DIRECTORY.md
-          git commit -am "updating DIRECTORY.md" ||  true
-          git push --force origin HEAD:$GITHUB_REF || true
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Build directory
+        uses: TheAlgorithms/scripts/directory_md@main
```
Comment (tjgurwara99) [inline .github/workflows/update_directory_md.yml:26]: Hey 👋🏼 could one of you create a branch which uses the patch I made in [this PR](https://github.com/TheAlgorithms/scripts/pull/29).

Just change the above line to something like this and it should hopefully work - if it doesn't, ping me directly please and I can take a look at the action log

```suggestion
        uses: TheAlgorithms/scripts/directory_md@e24e3ba9cf7e90c91ac87f953593426328b86563
```

If it works then I can merge my fix and then you won't have to merge this suggested change.

**Pull request merged**